### ISSUE TITLE
Request: control elements

### ISSUE THREAD

**Issue by vingerha** [muppet3000/homeassistant-grott#22]
Hi, recently the NL solar power users were quite heavily punished by supplying elec. back in the net...at a neative price some lost 10's of euros over the weekend. 
I found this link and would like to suggest to see if this seto of (a few) controls can be added to the integration

https://community.home-assistant.io/t/growatt-inverter-on-off-pause/588275/3

**Comment by muppet3000** [comment]
Hi @vingerha thanks for the info.
This is definitely something that I have on the roadmap. Right now I'm trying to work out the last few bugs and problems and then I'll start working on configuration.

I believe grott has some functionality that will support this, I just haven't had time to investigate it myself.

I've got another project that uses the API to do some configuration but it was for AC charging rather than blocking export.

How do you stop yourself from exporting anyway? The power has to go somewhere!

**Comment by vingerha** [comment]
It is indeed a VERY good question... the power does enter the inverter and then to say 'gimme less'  is not how it works I guess. Will ask the poster too

**Comment by muppet3000** [comment]
The power has to go 'somewhere', the only places that it can go are:
1. The house (the 'load')
2. Batteries - charging batteries
3. The grid

So you have to make sure that the priorities are set correctly but at the end of the day, if you're generating more energy than your house is consuming it has to go out to the grid.

Either way - I will be adding configuration to this integration at some point, hopefully soon!

**Comment by vingerha** [comment]
In this case, it is the idea to not (!) go to the grid as this resulted in hefty costs last weekend.

**Comment by muppet3000** [comment]
Yeah I understand, but where else can it go, other than turning all the appliances in your house on and making sure any batteries you have are filled?

**Comment by ablansinger** [comment]
> The power has to go 'somewhere', the only places that it can go are:
> 
>     1. The house (the 'load')
> 
>     2. Batteries - charging batteries
> 
>     3. The grid
> 
> 
> So you have to make sure that the priorities are set correctly but at the end of the day, if you're generating more energy than your house is consuming it has to go out to the grid.
> 
> Either way - I will be adding configuration to this integration at some point, hopefully soon!

The Inverters typically have a tracker which optimse the current/voltage to provide optimal power. If less power is required the inverter can simply increase or decrease the current to a less optimal point and thus less power is produced and less power needs to be distributed.

**Comment by vingerha** [comment]
Please read the ticket... the idea is to block going it to the grid if certain (price) conditions apply. If one can remotely switch on/off the inverter for a certain period of time then this would be OK too I guess

**Comment by muppet3000** [comment]
The question is, what are you able to configure from the shinephone app? If you're able to do it from the app we'd be able to work out how to do it in HA (somehow)

**Comment by ablansinger** [comment]
> The question is, what are you able to configure from the shinephone app? If you're able to do it from the app we'd be able to work out how to do it in HA (somehow)

What I am normally doing from the app or from the growatt web, is that I turn off all "Grif first periods" then I set the export limit on and to 0%, and when the price rises then I turn off the Export limit and enable any Grid first periods I might have. My inverter will not allow grid first periods and export limit at the same time.
This should be possible to automate.
